### ISSUE TITLE
Replace `require_chef_omnibus` in all `kitchen.yml` files due to deprecations

### ISSUE THREAD

**Issue by jmery** [chef-cft/bjc#678]
The TK files for developing the demo throw deprecation warnings about `require_chef_omnibus`:

```
$$$$$$ Deprecated configuration detected:
require_chef_omnibus
Run 'kitchen doctor' for details.
```

`kitchen doctor` then returns the following:

```
-----> The doctor is in
       **** require_chef_omnibus deprecated
The 'require_chef_omnibus' attribute with version values will change
to use the new 'product_version' attribute.

Note: 'product_name' must be set in order to use 'product_version'
until 'product_name' replaces 'require_chef_omnibus' as the default.

# New Usage #
provisioner:
  product_name: <chef or chefdk>
  product_version: 13.3.42
```

TL;DR - Replace `require_chef_omnibus` in all `kitchen.yml` files with `product_name` + `product_version`
